### ISSUE TITLE
Wrong schema while using discriminator

### ISSUE THREAD

**Issue by alphawolf1988** [micronaut-projects/micronaut-openapi#114]
It seems to be, if i'm using the discriminator, the generated schema is wrong. 

```
    BrochurePageTracking:
      allOf:
      - $ref: '#/components/schemas/CoreDomainTracking'
      - required:
        - brochurePage
        type: object
        properties:
          brochurePage:
            maximum: 999
            minimum: 1
            type: integer
            description: The page of the brochure.
            format: int32
```
The "required" field ist listed under the "allOf" field.

**Comment by croudet** [comment]
What version are your using?

**Comment by alphawolf1988** [comment]
@croudet I'm using the micronaut-openapi version 1.3.3 and the micronaut-core version 1.2.7.
I've testet previous versions until micronaut-openapi 1.2.3, which introduced the problem (1.2.2 works, but doesn't support discriminator mappings).

**Comment by croudet** [comment]
Looks correct to me.
Check https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/
```yaml
    components:
      schemas:
        BasicErrorModel:
          type: object
          required:
            - message
            - code
          properties:
            message:
              type: string
            code:
              type: integer
              minimum: 100
              maximum: 600
        ExtendedErrorModel:
          allOf:     # Combines the BasicErrorModel and the inline model
            - $ref: '#/components/schemas/BasicErrorModel'
            - type: object
              required:
                - rootCause
              properties:
                rootCause:
                  type: string
```

**Comment by alphawolf1988** [comment]
@croudet the "required" has a leading dash. In the spec this is not correct, but maybe the solved ticket #117 fixed the problem. I will test it if there is a new release available.

**Comment by croudet** [comment]
The properties are just not in the same order, it doesn't matter.
